src/recipes/ai-context/index.ts

```diff
@@ -48,78 +70,171 @@ const promptForPath = async (): Promise<string> => {
+  )
+  return match ? match.key : null
+}
+
+/**
+ * Receives a process ID (pid) and returns both the command that the process was run with and its parent process ID. If the process is a known IDE, also returns information about that IDE.
+ */
+const getCommandAndParentPID = async (
+  pid: number,
+): Promise<{
+  parentPID: number
+  command: string
+  consumerKey: string | null
+}> => {
+  const { stdout } = await execa('ps', ['-p', String(pid), '-o', 'ppid=,comm='])
+  const output = stdout.trim()
+  const spaceIndex = output.indexOf(' ')
+  const parentPID = output.substring(0, spaceIndex)
+  const command = output.substring(spaceIndex + 1).toLowerCase()
+  return {
+    parentPID: parseInt(parentPID, 10),
+    command: command,
+    consumerKey: getConsumerKeyFromCommand(command),
+  }
+}
+
+const getPathByDetectingIDE = async (): Promise<ConsumerConfig | null> => {
+  // Go up the chain of ancestor process IDs and find if one of their commands matches an IDE.
+  const ppid = process.ppid
+  let result: Awaited<ReturnType<typeof getCommandAndParentPID>>
+  try {
+    result = await getCommandAndParentPID(ppid)
+    while (result.parentPID !== 1 && !result.consumerKey) {
+      result = await getCommandAndParentPID(result.parentPID)
+    }
+  } catch {
+    // The command "ps -p {pid} -o ppid=,comm=" didn't work,
+    // perhaps we are on a machine that doesn't support it.
+    return null
+  }
+
+  if (result?.consumerKey) {
+    const contextConsumer = cliContextConsumers.find((consumer) => consumer.key === result.consumerKey)
+    if (contextConsumer) {
+      return contextConsumer
+    }
+  }
+
+  return null
 }
 
 export const run = async ({ args, command }: RunRecipeOptions) => {
-  // Start the download in the background while we wait for the prompts.
-  const download = downloadFile(version).catch(() => null)
+  let consumer: ConsumerConfig | null = null
+  const filePath: string | null = args[0]
+
+  if (filePath) {
+    consumer = { ...rulesForDefaultConsumer, path: filePath }
+  }
 
-  const filePath = args[0] || (await promptForPath())
-  const { contents: downloadedFile, minimumCLIVersion } = (await download) ?? {}
+  if (!consumer && process.env.AI_CONTEXT_SKIP_DETECTION !== 'true') {
+    consumer = await getPathByDetectingIDE()
+  }
 
-  if (!downloadedFile) {
-    return logAndThrowError('An error occurred when pulling the latest context files. Please try again.')
+  if (!consumer) {
+    consumer = await promptForContextConsumerSelection()
   }
 
-  if (minimumCLIVersion && semver.lt(version, minimumCLIVersion)) {
-    return logAndThrowError(
-      `This command requires version ${minimumCLIVersion} or above of the Netlify CLI. Refer to ${chalk.underline(
-        'https://ntl.fyi/update-cli',
-      )} for information on how to update.`,
-    )
+  if (!consumer?.contextScopes) {
+    log('No context files found for this consumer. Try again or let us know if this happens again.')
```

Comment:
Should we add some instructions for _how_ they should let us know if this happens again?